### ISSUE TITLE
Default share owner 

### ISSUE THREAD

**Issue by tronder88** [rockstor/rockstor-core#1255]
Hi, just a suggestion,
is it possible to change default share owner and group from root/root to admin user created?

**Comment by MFlyer** [comment]
Hi @tronder88 , why do you wanna this??? Just to understand

Flyer

**Comment by tronder88** [comment]
Most of the shares I make is for docker images. And I do not want to use root as owner since then I must use root user for dockers. In addition, since most users are home users, it is very often used same group for all users, which will be admin group (GID1000).

And to just easen the folder creation, and reduce the use of root, that is why I suggest this. If there is a good reason for not doing it this way, pleas let me know and we can close thsi suggestion.

**Comment by MFlyer** [comment]
Don't think there's a real reason for root owning shares (that's why you can modify that, ex. I've got owner to active directory Admin), so changing it to rockstor created user instead of root should be ok.



@schakrava ??

**Comment by schakrava** [comment]
Perhaps we should let the user optionally specify a owner and group as part of Share creation? The default should however be root for simplicity. What do you guys think?

@phillxnet

**Comment by ScarabMonkey** [comment]
I agree that a prompt on Share creation would make sense - with a default already filled-in to make it easier.

**Comment by phillxnet** [comment]
@schakrava @ScarabMonkey 
I like the idea of optionally specifying owner and group during share creation and hopefully it will help users to be more mindful of such things and do more share separation by using users. Maybe we could also default to more strict permissions also ie 750 so that all shares are not world readable unless specifically requested to be so.

@schakrava 
I do agree with not using admin as default user and sticking with root. I would say admin is the last user we should have owning any shares, especially by default.

@MFlyer 
The root user owns all shares in a sense anyway so when we default to root we are not adding anything without the users intervention / knowledge, hence I like the idea that it stays as a conscious choice to elect a specific owner.

@tronder88 I think I get your concern and it's good you have brought this up as I think a pain point for NAS is user / rights management and it is very common to just do a 777 and call it a day; we can't encourage this as it is just plain wrong. But I don't agree that admin is the way to go for owner or group, plus each user defaults to having their own group so it's best we encourage the creation of users and groups for natural separation. Plus the admin users in Rockstor necessarily have privileges beyond those of other users so we just need to encourage people to make and use different users and anything that makes this smoother is a good thing so I definitely agree with your sentiment on this one. I have tried to emphasise user creation and awareness of access rights in my various documentation contributions also. 

.
